client/pages/verify.js

```diff
@@ -0,0 +1,218 @@
+import React from 'react'
+// import PropTypes from 'prop-types'
+
+// Component imports
+import { actions, connect, actionStatus } from '../store'
+import { Router, Link } from '../routes'
+import PageWrapper from '../components/PageWrapper'
+import httpStatus from '../helpers/httpStatus'
+import PasswordField from '../components/PasswordField'
+import { passwordPattern } from '../data/RegExpr'
+
+
+// Component Constants
+
+@connect
+class Verify extends React.Component {
+  /***************************************************************************\
+    Class Properties
+  \***************************************************************************/
+
+  state = {
+    password: '',
+    submitted: false,
+    submitting: false,
+    validating: false,
+  }
+
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+  _handleSubmit = async (event) => {
+    const {
+      password,
+      token,
+    } = this.state
+
+    event.preventDefault()
+
+    this.setState({ submitting: true })
+
+    const resetResponse = await this.props.resetPassword({
+      password,
+      token,
+    })
+
+    if (resetResponse.status === actionStatus.SUCCESS) {
+      this.setState({
+        submitted: true,
+        submitting: false,
+      })
+      Router.pushRoute('home', { authenticate: true })
+    }
+  }
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+
+  static async getInitialProps ({ query, store, res }) {
+    // eslint-disable-next-line id-length
+    const { type, t: token } = query
+    let destination = null
+    let response = null
+
+    switch (type) {
+      case 'reset':
+        console.log('Reset Redirect', token)
+        response = await actions.verifyResetToken(token)(store.dispatch)
+        if (response.status === actionStatus.SUCCESS) {
+          destination = null
+        }
+        break
+      case 'email':
+        console.log('Verifying Email')
+        response = await actions.verifyEmailToken(token)(store.dispatch)
+        if (response.status === actionStatus.SUCCESS) {
+          destination = '/profile'
+        }
+        break
+      case 'session':
+        console.log('Session was recieved')
+        response = await actions.verifySessionToken(token)(store.dispatch)
+        if (response.status === actionStatus.SUCCESS) {
+          destination = '/?authenticate=true'
+        }
+        break
+      default:
+        destination = '/'
+        break
+    }
+    if (destination === null) {
+      return response
+    }
+    if (res) {
+      res.writeHead(httpStatus.FOUND, {
+        Location: `${destination}`,
+      })
+      res.end()
+      res.finished = true
+    } else {
+      Router.replace(`${destination}`)
+    }
+
+    return { response }
```

Comment:
you should only return what you need when working in `getInitialProps`.

In this case you only need the status so you could pass it back like:
```suggestion
    return {
      tokenIsValid: response.status
    }
```
and then use it as a normal prop.
```javascript
const { tokenIsValid } = this.props
```